Code/GraphMol/Fingerprints/AtomPairs.cpp

```diff
@@ -444,7 +444,7 @@ SparseIntVect<boost::int64_t> *getTopologicalTorsionFingerprint(
         // look for a cycle that doesn't start at the first atom
         // we can't effectively canonicalize these at the moment
         // (was github #811)
-        if (pIt != path.begin() && pAtoms[*pIt]) {
+        if (pIt != path.begin() && *pIt != *(path.begin()) && pAtoms[*pIt]) {
```

Comment:
Not blocking, but isn't this just?

```
if (*pIt != *(path.begin()) && pAtoms[*pIt])
```

I did the same thing internally when I looked at it so I also didn't notice the issue :)